tests/model_test.py

```diff
@@ -0,0 +1,248 @@
+    )
+    assert sc.isclose(
+        chopper1.tofs.visible.data.coords['tof'][1],
+        (0.5 * (topen + tclose)).to(unit='us'),
+    )
+    assert sc.isclose(
+        chopper1.tofs.blocked.data.coords['tof'][0], (1.1 * tclose).to(unit='us')
+    )
+    assert len(chopper2.tofs.visible) == 1
+    # Blocks only one neutron, the other is blocked by chopper1
+    assert len(chopper2.tofs.blocked) == 1
+    assert sc.isclose(
+        chopper2.tofs.visible.data.coords['tof'][0],
+        (pulse.wavelengths[1] * chopper2.distance * tof.utils.alpha).to(unit='us'),
+    )
+    assert sc.isclose(
+        chopper2.tofs.blocked.data.coords['tof'][0],
+        (pulse.wavelengths[0] * chopper2.distance * tof.utils.alpha).to(unit='us'),
+    )
+    assert len(detector.tofs.visible) == 1
+    assert sc.isclose(
+        detector.tofs.visible.data.coords['tof'][0],
+        (pulse.wavelengths[1] * detector.distance * tof.utils.alpha).to(unit='us'),
+    )
+
+
+def test_two_choppers_one_and_two_openings():
+    topen = 5.0 * ms
+    tclose = 16.0 * ms
+    chopper1 = make_chopper(
+        topen=[topen],
+        tclose=[tclose],
+        f=10.0 * Hz,
+        phase=0.0 * deg,
+        distance=10 * meter,
+        name="chopper1",
+    )
+
+    chopper2 = make_chopper(
+        topen=[9.0 * ms, 15.0 * ms],
+        tclose=[12.0 * ms, 20.0 * ms],
+        f=15.0 * Hz,
+        phase=0.0 * deg,
+        distance=15 * meter,
+        name="chopper2",
+    )
+
+    detector = tof.Detector(distance=20 * meter, name="detector")
+
+    # Make a pulse with 7 neutrons:
+    # - 2 neutrons blocked by chopper1
+    # - 3 neutrons blocked by chopper2
+    # - 2 neutrons detected (one through each of chopper2's openings)
+    pulse = make_pulse(
+        arrival_times=sc.concat(
+            [
+                0.9 * topen,
+                1.01 * topen,
+                1.3 * topen,
+                1.8 * topen,
+                0.5 * (topen + tclose),
+                0.85 * tclose,
+                1.1 * tclose,
+            ],
+            dim='event',
+        ),
+        distance=chopper1.distance,
+    )
+
+    model = tof.Model(pulse=pulse, choppers=[chopper1, chopper2], detectors=[detector])
+    model.run()
+
+    assert len(chopper1.tofs.visible) == 5
+    assert len(chopper1.tofs.blocked) == 2
+    assert len(chopper2.tofs.visible) == 2
+    assert len(chopper2.tofs.blocked) == 3
+    assert len(detector.tofs.visible) == 2
+
+
+def test_neutron_conservation():
+    N = 100_000
+    pulse = tof.Pulse(neutrons=N, kind='ess')
+
+    chopper1 = make_chopper(
+        topen=[5.0 * ms],
+        tclose=[16.0 * ms],
+        f=10.0 * Hz,
+        phase=0.0 * deg,
+        distance=10 * meter,
+        name="chopper1",
+    )
+    chopper2 = make_chopper(
+        topen=[9.0 * ms, 15.0 * ms],
+        tclose=[15.0 * ms, 20.0 * ms],
+        f=15.0 * Hz,
+        phase=0.0 * deg,
+        distance=15 * meter,
```

Comment:
Could parametrize this, or consider using hypothesis in the future?

Reply:
ok for hypothesis in the future

Reply:
See #25